src/wallet/spend.cpp

```diff
@@ -204,9 +230,47 @@ CoinsResult AvailableCoins(const CWallet& wallet,
             if (!spendable && only_spendable) continue;
 
             int input_bytes = GetTxSpendSize(wallet, wtx, i, (coinControl && coinControl->fAllowWatchOnly));
-            result.coins.emplace_back(outpoint, output, nDepth, input_bytes, spendable, solvable, safeTx, wtx.GetTxTime(), tx_from_me, feerate);
+            // When parsing a scriptPubKey, Solver returns the parsed pubkeys or hashes (depending on the script)
+            // We don't need those here, so we are leaving them in return_values_unused
+            std::vector<std::vector<uint8_t>> return_values_unused;
+            TxoutType type;
+            bool is_from_p2sh{false};
+
+            if (wtx.tx->vout[i].scriptPubKey.IsPayToScriptHash()) {
+                CScript redeemScript;
+                CTxDestination destination;
+                ExtractDestination(wtx.tx->vout[i].scriptPubKey, destination);
+                const CScriptID& hash = CScriptID(std::get<ScriptHash>(destination));
+                provider->GetCScript(hash, redeemScript);
```

Comment:
added a check on `spendable` before looking up the CScript (since it doesn't make sense to try and find the script if it's not spendable by us) and handled the case where ExtractDest fails or GetScript fails with a `continue`.